eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Gov.hs

```diff
@@ -198,6 +207,21 @@ noSuchGovActions gas gaIds =
    in failureUnless (Set.null unknownGovActionIds) $
         GovActionsDoNotExist unknownGovActionIds
 
+checkVotesAreValid ::
+  ConwayEraPParams era =>
+  Map.Map (GovActionId (EraCrypto era)) (GovActionState era) ->
+  VotingProcedures era ->
+  Test (ConwayGovPredFailure era)
+checkVotesAreValid gas (VotingProcedures vps) =
```

Comment:
@aniketd Here is a challenge for you. Can you reimplement this function in such a way that whenever validation is not enabled there will be no computation.

In other words the whole validation for all of the votes have to be encapsulated in the first argument to the `failureUnless`:
```haskell
failureUnless (Map.null votes) $ DisallowedVotes votes
```

where

```haskell
votes :: Map (Voter c) (Set (GovActionId c))
```


This means that `VoterMismatch (Voter (EraCrypto era)) (GovActionId (EraCrypto era))` will be replaced by `DisallowedVotes !(Map (Voter c) (Set (GovActionId c)))`

**NOTE**: You can also reuse the argument that is supplied to `noSuchGovActions` function

Reply:
I think you missed my note: 

> You can also reuse the argument that is supplied to noSuchGovActions function

Reply:
I used the same arguments as those given to `noSuchGovActions`, but the comments are stray. Just cleaned them up and pushed again.